sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/BuiltInFunctionResource.java

```diff
@@ -0,0 +1,151 @@
+
+  @Inject WriteDocumentsService documentWriteService;
+
+  @Inject TableManager tableManager;
+
+  @Operation(
+      description =
+          "Execute a built-in function (e.g. $push and $pop) against a value in this document. Performance may vary.")
+  @Parameters(
+      value = {
+        @Parameter(name = "namespace", ref = OpenApiConstants.Parameters.NAMESPACE),
+        @Parameter(
+            name = "collection",
+            ref = OpenApiConstants.Parameters.COLLECTION,
+            description = "The collection of the document. Will be created if it does not exist."),
+        @Parameter(name = "document-id", ref = OpenApiConstants.Parameters.DOCUMENT_ID),
+      })
+  @RequestBody(ref = OpenApiConstants.RequestBodies.WRITE_BUILT_IN_FUNCTION)
+  @APIResponses(
+      value = {
+        @APIResponse(
+            responseCode = "200",
+            description = "Function is executed. Certain functions will return a value.",
+            content = {
+              @Content(
+                  schema =
+                      @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                          properties = {
+                            @SchemaProperty(
+                                name = "documentId",
+                                type = SchemaType.STRING,
+                                description = "The ID of the written document."),
+                            @SchemaProperty(
+                                name = "profile",
+                                implementation = ExecutionProfile.class,
+                                nullable = true),
+                          }),
+                  examples = @ExampleObject(ref = OpenApiConstants.Examples.DOCUMENT_SINGLE))
+            }),
+        @APIResponse(
+            responseCode = "404",
+            description = "Not found.",
+            content =
+                @Content(
+                    examples = {
+                      @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST)
+                    },
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @POST
+  @Path("{collection:\\w+}/{document-id}/{document-path:.*}/function")
+  public Uni<RestResponse<Object>> executeBuiltInFunction(
+      @Context UriInfo uriInfo,
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection,
+      @PathParam("document-id") String documentId,
+      @PathParam("document-path") List<PathSegment> documentPath,
+      @QueryParam("ttl") Integer ttl,
```

Comment:
Is `ttl` actually part of the function execution? We need the validation never the less.

Reply:
`ttl` definitely not used, by `ttlAuto` might be required here?